subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultCacheAccess.java

```diff
@@ -326,14 +353,12 @@ private void afterLockAcquire(FileLock fileLock) {
         assert this.fileLock == null;
         this.fileLock = fileLock;
         this.stateAtOpen = fileLock.getState();
-        takeOwnershipNow();
-        try {
+
+        withOwnershipNow(() -> {
```

Comment:
The only thing that really sticks out here is that these two methods never acquired the `stateLock` and now they do. Not sure what the implications of that is though.

Reply:
Seems that should be an okay change then